### ISSUE TITLE
Installation with docker-compose on Fedora CoreOS fails with Nextcloud versions greater than 18

### ISSUE THREAD

**Issue by albertaparicio** [nextcloud/docker#1335]
I am trying to install Nextcloud with docker-compose on Fedora CoreOS. I am using the FPM configuration with Alpine images. I have taken the docker-compose.yml from [here](https://github.com/nextcloud/docker/blob/master/.examples/docker-compose/insecure/postgres/fpm/docker-compose.yml) as a starting point. 

I am facing an error when the apps are installed the first time I run the containers.

**I have attached a minimal working example of the configuration I am using [HERE](https://github.com/nextcloud/docker/files/5721255/minimal_working_example.zip)**

If the tag for Nextcloud points to a version greater than 18 I get this error and after several retries the install ultimately fails:

```
cloud_1  | Initializing nextcloud 20.0.3.2 ...
cloud_1  | Initializing finished
cloud_1  | New nextcloud instance
cloud_1  | Installing with PostgreSQL database
cloud_1  | starting nextcloud installation
cloud_1  | PostgreSQL username and/or password not valid
cloud_1  |  -> You need to enter details of an existing account.
cloud_1  | retrying install...
cloud_1  | An unhandled exception has been thrown:
cloud_1  | OC\DatabaseException: An exception occurred while executing 'SHOW SERVER_VERSION':
cloud_1  | 
cloud_1  | Failed to connect to the database: An exception occurred in driver: SQLSTATE[08006] [7] FATAL:  database "nextcloud" does not exist in /var/www/html/lib/private/legacy/OC_DB.php:73
cloud_1  | Stack trace:
cloud_1  | #0 /var/www/html/lib/private/legacy/OC_DB.php(143): OC_DB::prepare('SHOW SERVER_VER...', NULL, NULL)
cloud_1  | #1 /var/www/html/lib/private/legacy/OC_Util.php(984): OC_DB::executeAudited(Array)
cloud_1  | #2 /var/www/html/lib/private/legacy/OC_Util.php(964): OC_Util::checkDatabaseVersion()
cloud_1  | #3 /var/www/html/lib/private/Console/Application.php(161): OC_Util::checkServer(Object(OC\SystemConfig))
cloud_1  | #4 /var/www/html/console.php(99): OC\Console\Application->loadCommands(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
cloud_1  | #5 /var/www/html/occ(11): require_once('/var/www/html/c...')
cloud_1  | #6 {main}retrying install...
```

The log from the DB shows that the `nextcloud` database is not present:

```
db_1     | 2020-12-20 17:25:35.162 UTC [50] FATAL:  database "nextcloud" does not exist
```

If the tag is `18-fpm-alpine` Nextcloud installs with no issues at all. If I then change the version to `19-fpm-alpine` and restart the containers it updates well. The same if I change the tag again from `19-fpm-alpine` to `20-fpm-alpine`.

However, if Nextcloud is to be installed directly with `19-fpm-alpine` or `20-fpm-alpine`, the installation fails.

Could anybody point out the issue with this? 

Is there a bug in the images for versions 19 and higher? Is there anything wrong in my configuration?

Thank you

**Comment by DudleyDursley** [comment]
The database name in the postgress container might be wrong. Try POSTGRES_DB=nextcloud otherwise the database name should default to POSTGRES_USER.

There were a few problems with the database setup a while ago where nextcloud would always create an additional database if I remeber correctly. That might make your 18 install work but got fixed later.

**Comment by albertaparicio** [comment]
Thank you @DudleyDursley for your quick response

I have tried putting `POSTGRES_DB=nextcloud` in the postgres container. With this the installation succeeded with Nextcloud 20

However, right after the install finishes I get [this other error](https://github.com/nextcloud/docker/files/5723803/error.txt):

```
cloud_1  | setting trusted domains…
cloud_1  | An unhandled exception has been thrown:
cloud_1  | PDOException: SQLSTATE[42501]: Insufficient privilege: 7 ERROR:  permission denied for table oc_appconfig in /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:78
cloud_1  | Stack trace:
cloud_1  | #0 /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php(78): PDO->query('SELECT * FROM "...')
cloud_1  | #1 /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(911): Doctrine\DBAL\Driver\PDOConnection->query('SELECT * FROM "...')
cloud_1  | #2 /var/www/html/lib/private/DB/Connection.php(202): Doctrine\DBAL\Connection->executeQuery('SELECT * FROM "...', Array, Array, NULL)
cloud_1  | #3 /var/www/html/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php(206): OC\DB\Connection->executeQuery('SELECT * FROM "...', Array, Array)
[...]
```

See full trace [HERE](https://github.com/nextcloud/docker/files/5723807/error.txt)

Maybe the database name is not the only required change?

Thank you

**Comment by DudleyDursley** [comment]
I'm not using `postgres` as database username. That is the default an might come with some special configuration, I don't know. Otherwise I'm out of ideas.

**Comment by albertaparicio** [comment]
I have tried out your suggestion and I managed to get it working

However, the trick was not changing the username or the DB name.

The trick was to first start the DB container on its own (`docker-compose up db`), so it can set itself up. Once the DB is ready, shut it down (`docker-compose down`) and then start all the services (`docker-compose up`).

From what I can tell, it looks like the issue is more of a race condition between containers, rather than a bug or a code issue. It looks like if I start all services at once so they get setup for the first time, Nextcloud tries to send queries to the DB before it is ready, it goes into a weird state and the installation fails. However if I set the DB first and then install Nextcloud it works well.

@DudleyDursley Thank you for your tips. I will close the issue now
